### PR TITLE
Adapt hawk_test code to support testing for 12-SP2

### PR DESCRIPTION
**Pull request by ricardobranco777**
Adapt hawk_test code to support testing for 12-SP2

Also fix Dockerfile for best & recommended practices (avoid consecutive RUN's that increase image size).

### CODE REVIEW

File: e2e_test/hawk_test_driver.py [hunk 1]
```diff
@@ -533,9 +541,17 @@ def test_click_around_edit_conf(self):
...
-        self.check_and_click_by_xpath("while checking around edit configuration",
-                                      [Xpath.HREF_CONSTRAINTS, Xpath.HREF_NODES, Xpath.HREF_TAGS,
-                                       Xpath.HREF_ALERTS, Xpath.HREF_FENCING])
+
+        click_list = [
+            Xpath.HREF_CONSTRAINTS, Xpath.HREF_NODES, Xpath.HREF_TAGS,
+            Xpath.HREF_ALERTS, Xpath.HREF_FENCING
+        ]
+
+        # HAWK version below 12-SP3 does not provide link to fencing
```
Comment (MalloZup) [inline e2e_test/hawk_test_driver.py:550]: Why this? I'm thinking if we should not fix that.

**Review by MalloZup — APPROVED** [review]
looks good thx! just a minor question but is not inherent to the tests

File: e2e_test/hawk_test_driver.py (hunk 1 above)
Comment (ricardobranco777) [reply e2e_test/hawk_test_driver.py:550]: If your team plans to fix that we can drop this when it's fixed.

Comment (MalloZup) [reply e2e_test/hawk_test_driver.py:550]: sure, we need first to track/understand  the issue then we can fix it.

Comment (MalloZup) [reply e2e_test/hawk_test_driver.py:550]: https://github.com/ClusterLabs/hawk/issues/219 can you provide more feedback there?

File: e2e_test/hawk_test_driver.py [hunk 2]
```diff
@@ -533,9 +541,17 @@ def test_click_around_edit_conf(self):
...
+        if Version(self.test_version) < Version("12-SP3"):
+            click_list.remove(Xpath.HREF_FENCING)
+
+        self.check_and_click_by_xpath("while checking around edit configuration", click_list)
```
Comment (MalloZup) [inline e2e_test/hawk_test_driver.py:554]: while checking around is a bit to much familiar but I guess is ok, :grin:

**Review by MalloZup — APPROVED** [review]

**Comment by MalloZup** [comment]
@ricardobranco777  can you send a PR to `sle12` and `factory` branch backporting this? 

https://github.com/ClusterLabs/hawk/blob/master/doc/release.md

**Pull request merged**